src/adapter/src/catalog/state.rs

```diff
@@ -824,6 +826,8 @@ impl CatalogState {
         custom_logical_compaction_window: Option<CompactionWindow>,
     ) -> Result<CatalogItem, AdapterError> {
         let mut session_catalog = self.for_system_session();
+        session_catalog.set_planning_id(id);
```

Comment:
Can this get stashed in a better place? How about the `PlanContext`? Still accessible via the StatementContext where you use it later on.

Reply:
Good call--I'm so used to just sticking everything in `SessionCatalog`.